README.md

```diff
@@ -621,6 +621,17 @@ Here are the metrics settings:
   May be set to "none" (the default in older [variants](variants/), up through aws-k8s-1.19), "integrity" (the default for newer [variants](variants/)), or "confidentiality".
   **Important note:** this setting cannot be lowered (toward 'none') at runtime.
   You must reboot for a change to a lower level to take effect.
+* `settings.kernel.modules.<name>.allowed`: Whether the named kernel module is allowed to be loaded.
```

Comment:
Have you considered having a `blocked` list? The current API implementation doesn't allow us to delete keys in the settings so if I want to allow a module in the current scheme, I have to set the `allowed` flag as true, which practically has the same effect as not having the setting at all. With this scheme, users could end up with many keys that they might not really need if they later decide to allow some of the previously blocked modules. But with the `blocked` list, they only have to modify one list, instead of many keys.

Also with a list, you can save yourself the extra `if` and `unless` statements in the handlebars template:

```handlebars
{{#each settings.kernel.modules.blocked}}
install {{this}} /bin/true
{{/each}}
```

And probably the extra `KmodSetting` type.

Reply:
I did consider it. The problem with lists is that they can't be incrementally updated; you have to reapply the current state with the desired additions or removals to get to the desired state. That makes it difficult to document independent steps for disabling individual modules, which is something I had to do for the CIS benchmark.